### PR TITLE
Make requests smaller or equal to limits

### PR DESCRIPTION
**Pull request by tkulczynski**
This also extracts common code from updater & admission controller to a library.

### CODE REVIEW

File: vertical-pod-autoscaler/admission-controller/logic/recommendation_provider.go [hunk 1]
```diff
@@ -41,76 +41,20 @@ func NewRecommendationProvider(vpaLister vpa_lister.VerticalPodAutoscalerLister)
 	return &recommendationProvider{vpaLister: vpaLister}
 }
 
-// getRecomendedResources overwrites pod resources Request field with recommended values.
+// getRecomendedResources returns the recommended resources Request for each container in the given pod.
```
Comment (kgrygiel) [inline vertical-pod-autoscaler/admission-controller/logic/recommendation_provider.go]: "in the same order they are specified in the pod.Spec"

File: vertical-pod-autoscaler/pkg/utils/vpa/capping.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+// GetCappedRecommendationForContainer returns a recommendation extracted for the given container, adjusted to obey policy and limits.
+func GetCappedRecommendationForContainer(
+	container apiv1.Container,
+	podRecommendation *vpa_types.RecommendedPodResources,
+	policy *vpa_types.PodResourcePolicy) (apiv1.ResourceList, error) {
+	containerRecommendation := getRecommendationForContainer(podRecommendation, container)
+	if containerRecommendation == nil {
+		return nil, fmt.Errorf("no recommendation available for container name %v", container.Name)
+	}
+	capRecommendationToContainerLimit(containerRecommendation, container)
```
Comment (kgrygiel) [inline vertical-pod-autoscaler/pkg/utils/vpa/capping.go]: I think capping to the limit should be applied *after* the VPA policy. It normally shouldn't matter, but in the unlikely case of a conflict (currently: minAllowed > limit), capping to limit should take precedence. I guess the updater could set some error condition in such case (maybe a TODO for now?)

**Comment by tkulczynski** [comment]
Applied all your comments, removed the glogs from vpa/capping.go completely. PTAL

**Comment by tkulczynski** [comment]
@mwielgus could you please merge?

**Pull request merged**